File: day7/day7.py
file1 = open('day7.txt', 'r') 
Lines = file1.readlines() 
  
def part1():
    count = 0
    par = ["shiny gold"]
    search = True
    while(search):
        mark = True
        for bag in par:
            for line in Lines: 
                if bag in line: 
                    toIn  = line[0:line.find("bags")-1];
                    if toIn not in par:
                        par.append(toIn)
                else:
                    mark = False;
        if not mark:
            search = False;
    return len(par)-1

def part2():
    for line in Lines:
        if line[0:line.find("bags")-1] == "shiny gold":
            start = line[line.find("contain")+8:].replace("bags","")
    
    for bag in start

    print(start)
    

def main():
    print("Advent of Code!")
    print(part1())
    part2()
if __name__ == "__main__":
    main()

File: day7/day7.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <stdio.h>
#include <vector> 


using namespace std;

int main(void)
{
    int count = 0;
    vector<string> parent;
    string target = "shiny gold";
    fstream newfile;
    newfile.open("day7.txt",ios::in);
    
    if (newfile.is_open()){  
	string tp;
	string rule;
	size_t pos;
	while(getline(newfile, tp)){ 
	    pos = tp.find("bag");
	    rule  = tp.substr(0,pos);
	    if(tp.find(target) != string::npos){
		count++;
		

	    }
	}
	    
	cout << rule << "\n"; 
	newfile.close(); 
    } 
    cout << "------------------------------\n";
    newfile.open("day7.txt",ios::in);
    if(newfile.is_open()){
	cout << "Open";
	string tp2;
	while(getline(newfile, tp2)){
	    int len = parent.size();
	    for(int i = 0; i < len; i++){
		size_t pos = tp2.find("bag");
		string just = tp2.substr(pos);
		cout << just << " - " << parent[i]  << "\n";
		if(just.find(parent[i]) != string::npos){
		    count++;
		}
	    }
	}
	newfile.close();
    }

    cout << count;
    return 0;
}

